packages/app-folders/src/components/Tree/index.tsx

```diff
@@ -93,110 +30,42 @@ export const FolderTree: React.FC<Props> = ({
-
-    useDeepCompareEffect(() => {
-        if (folders) {
-            setTreeData(createTreeData(folders, focusedFolderId));
-            setInitialOpenList(createInitialOpenList(folders, focusedFolderId));
-        }
-    }, [Object.assign({}, folders), focusedFolderId]);
-
-    const handleDrop = async (
-        newTree: NodeModel<DndItemData>[],
-        { dragSourceId, dropTargetId }: DropOptions
-    ) => {
-        try {
-            const item = folders.find(folder => folder.id === dragSourceId);
 
-            if (!item) {
-                throw new Error("Folder not found");
-            }
-
-            setTreeData(newTree);
-            await updateFolder({
-                ...item,
-                parentId: dropTargetId !== ROOT_ID ? (dropTargetId as string) : null
-            });
-        } catch (error) {
-            return showSnackbar(error.message);
-        }
-    };
+    // Little CSS trick here: since the folder title has absolute position, user can drag a folder over it and move it to root folder.
+    // While we are moving folders around we disable any title pointer event.
+    const [isDragging, setIsDragging] = useState<boolean>(false);
 
     return (
         <Container>
-            <Title title={title} onClick={onTitleClick} />
-
-            {folders && folders.length > 0 && (
-                <DndProvider backend={MultiBackend} options={getBackendOptions()}>
-                    <Tree
-                        tree={treeData}
-                        rootId={ROOT_ID}
-                        onDrop={handleDrop}
-                        sort={false}
-                        render={(node, { depth, isOpen, onToggle }) => (
-                            <Node
-                                node={node}
-                                depth={depth}
-                                isOpen={isOpen}
-                                onToggle={onToggle}
-                                onClick={data => onFolderClick(data)}
-                                onUpdateFolder={data => {
-                                    setSelectedFolder(data);
-                                    setUpdateDialogOpen(true);
-                                }}
-                                onDeleteFolder={data => {
-                                    setSelectedFolder(data);
-                                    setDeleteDialogOpen(true);
-                                }}
-                            />
-                        )}
-                        dragPreviewRender={monitorProps => (
-                            <NodePreview monitorProps={monitorProps} />
-                        )}
-                        classes={{
-                            root: "treeRoot",
-                            dropTarget: "dropTarget"
-                        }}
-                        initialOpen={initialOpenList}
+            <Title title={title} onClick={onTitleClick} isDragging={isDragging} />
+            {!folders ? (
```

Comment:
I'd say, for better readability, we do this above the `return` line.

Having ternary and then nested ternary operators is a bit hard to read.

Reply:
Done it, thank you!